weave

```diff
@@ -138,6 +139,38 @@ collect_ip_args() {
     fi
 }
 
+dns_available() {
+    [ "$WITHOUT_DNS" = "true" ] && return 1
+    [ "$WITH_DNS" = "true" ] && return 0
+    "$@"
+}
+
+collect_dns_args() {
+    WITH_DNS=false
+    WITHOUT_DNS=false
+    DNS_ARG_COUNT=0
+    while [ $# -gt 0 ] ; do
+        case "$1" in
+            --with-dns)
+                WITH_DNS=true
+                DNS_ARG_COUNT=$((DNS_ARG_COUNT + 1))
+                shift
+                ;;
+            --without-dns)
+                WITHOUT_DNS=true
+                DNS_ARG_COUNT=$((DNS_ARG_COUNT + 1))
+                shift
+                ;;
+            *)
+                break
+                ;;
+        esac
+    done
+    if [ "$WITH_DNS" = "true" -a "$WITHOUT_DNS" = "true" ] ; then
+        usage
+    fi
+}
```

Comment:
oops. accidentally deleted my comments here...

We don't need the looping, or checking whether both `--with` options were given. keep it simple.